internal/policy/policy.go

```diff
@@ -675,23 +674,25 @@ func (s *State) findDelegationEntry(roleName string) (tuf.Delegation, error) {
 		delegationTargetsMetadata[name] = targetsMetadata
 	}
 
-	delegationsQueue := topLevelTargetsMetadata.Delegations.Roles
+	delegationsToCheck := topLevelTargetsMetadata.Delegations.Roles
 
 	for {
-		if len(delegationsQueue) == 0 {
+		if len(delegationsToCheck) == 0 {
 			return tuf.Delegation{}, ErrDelegationNotFound
 		}
 
-		delegation := delegationsQueue[0]
-		delegationsQueue = delegationsQueue[1:]
+		delegation := delegationsToCheck[0]
+		delegationsToCheck = delegationsToCheck[1:]
 
 		if delegation.Name == roleName {
 			return delegation, nil
 		}
 
 		if s.HasTargetsRole(delegation.Name) {
-			// TODO: clarifying terminating / reachability
-			delegationsQueue = append(delegationsQueue, delegationTargetsMetadata[delegation.Name].Delegations.Roles...)
+			if delegation.Terminating {
+				delegationsToCheck = delegationTargetsMetadata[delegation.Name].Delegations.Roles
```

Comment:
FYI: we've been discussing terminating delegation behavior in #168 and #206.

Reply:
@adityasaky Should I first create a PR just for testing this function, and later on, I can change this PR according to how termination is going to be handled?

Reply:
@adityasaky I have added test in #209

Reply:
Having looked at that, I think it can land regardless and we can work out terminating delegation in this context separately.